src/sync.cpp

```diff
@@ -112,11 +112,13 @@ static void potential_deadlock_detected(const std::pair<void*, void*>& mismatch,
         }
         LogPrintf(" %s\n", i.second.ToString());
     }
+    if (g_debug_lockorder_unittest) {
```

Comment:
is it me or this flag is never set?, only initialized to false.

Reply:
or wait.., following it now.
In mainnet this aborts and never get to the logic error point. Ok, nvm.